src/components/EditableField/EditableField.tsx

```diff
@@ -0,0 +1,169 @@
+import * as React from 'react'
+import propConnect from '../PropProvider/propConnect'
+import getValidProps from '@helpscout/react-utils/dist/getValidProps'
+import { classNames } from '../../utilities/classNames'
+import { noop } from '../../utilities/other'
+import { EditableFieldProps, EditableFieldState } from './EditableField.types'
+import {
+  EditableFieldUI,
+  LabelTextUI,
+  FieldActionsUI,
+  FieldButtonUI,
+} from './styles/EditableField.css'
+import { COMPONENT_KEY } from './EditableField.utils'
+import EditableFieldInput from './EditableFieldInput'
+import { key } from '../../constants/Keys'
+
+export class EditableField extends React.PureComponent<
+  EditableFieldProps,
+  EditableFieldState
+> {
+  static className = 'c-EditableField'
+  static defaultProps = {
+    innerRef: noop,
+    initialValue: '',
+    type: 'text',
```

Comment:
What do you think about using constants for values like this, like `inputTypes.TEXT` or something like that? I think its helpful to guard against typos but also can be helpful in testing to have object of constants that can be iterated over to produce inputs of all the different types.